### ISSUE TITLE
Using Ecoule for a static HTML page

### ISSUE THREAD

**Issue by mulderp** [gausby/ecoule#4]
As follow up to: https://github.com/gausby/ecoule/issues/3, was trying this:

```
 npm install -g ecoule
```

Then:

   → ecoule
   -bash: ecoule: command not found

I was reminded that this is a library rather than a command line tool, but still maybe it might be interesting. What I would need for a static page is generating a basic HTML layout. How could is create a static, empty HTML project?

**Comment by mulderp** [comment]
I do:

```
→ vim package.json   
// add "ecoule": "latest"

Fr Feb 21, 11:31:34
/Users/mulder/projects/github/statichtml master
→ npm install
npm WARN package.json statichtml@0.0.0 No README.md file found!
npm http GET https://registry.npmjs.org/ecoule
npm http 304 https://registry.npmjs.org/ecoule
npm http GET https://registry.npmjs.org/operandi/0.0.6
npm http GET https://registry.npmjs.org/pursuit/0.2.3
npm http GET https://registry.npmjs.org/ecoule-output-object-reference/0.0.1
npm http 304 https://registry.npmjs.org/pursuit/0.2.3
npm http 304 https://registry.npmjs.org/operandi/0.0.6
npm http 304 https://registry.npmjs.org/ecoule-output-object-reference/0.0.1
npm WARN engine operandi@0.0.6: wanted: {"node":">=0.10.5"} (current: {"node":"v0.10.4","npm":"1.2.18"})
ecoule@0.0.18 node_modules/ecoule
├── ecoule-output-object-reference@0.0.1
├── pursuit@0.2.3
└── operandi@0.0.6

Fr Feb 21, 11:31:39
/Users/mulder/projects/github/statichtml master
→ vim workflow.js
// add Ecoule workflow
```

Then, I would need an HTML transformer, or kind of template?

**Comment by gausby** [comment]
I think a command line tool would be a project of its own. This command line tool could, based on some strategies, created an ecoule config, that looked for a dot-file with settings that could be used to create the configuration.

A configuration auto-discover strategy chould be:
1. Look for a folder called .ecoule/ within the `cwd`
2. if not found; traverse the dictionary structure upwards, until a folder with a folder called .ecoule/ is found
3. if not found; look for the .ecoule/ folder in the ~
4. if not found; return an error message

I think an _ecoule-cli_-project would be in order for this.

**Comment by mulderp** [comment]
Yes, that sounds nice, maybe I can give it a try. What I saw too is: https://github.com/coderoshi/systatic

There, you get some basic templates too, basically an index.html and a 404.html. For my static pages I would like to have these too. Any thoughts where these fit?

**Comment by mulderp** [comment]
What is interesting too, is the command: `publish` - https://github.com/coderoshi/systatic/blob/master/bin/systatic#L42-L58

maybe it would be a nice starting point,

```
ecoule init
ecoule publish
```

**Comment by gausby** [comment]
`init` and `publish` would indeed be nice.

I haven't given the publish step too much thought. I was thinking the individual output, attached to a transformer, would be able to handle this step.

Ecoule is for more than just static websites. It could be used for building an ebook, an configuration file or something third. In some cases the `publish` step would just dump the result of the transformer into a folder on the hard drive.
